### PR TITLE
[CorfuStore] API to return CorfuStoreEntries, keySet & clear.

### PR DESCRIPTION
**Pull request by zalokhan**
## Overview

Description:
Added API to return key with the payload on scan and filter and queries.
Added API to return the keyset.
Added API to clear a table.

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/runtime/collections/CorfuStoreEntry.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import com.google.protobuf.Message;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+/**
+ * Created by zlokhandwala on 11/4/19.
+ */
+@EqualsAndHashCode
+public class CorfuStoreEntry<K extends Message, V extends Message, M extends Message> {
```
Comment (zalokhan) [inline runtime/src/main/java/org/corfudb/runtime/collections/CorfuStoreEntry.java]: Need to add comments.

File: runtime/src/main/java/org/corfudb/runtime/collections/Query.java [hunk 2]
```diff
@@ -163,6 +163,20 @@ public int count(@Nonnull final String tableName,
         }
     }
 
+    @Nonnull
+    private <K extends Message, V extends Message, M extends Message>
+    List<CorfuStoreEntry<K, V, M>> scanAndFilterByEntry(
```
Comment (hisundar) [inline runtime/src/main/java/org/corfudb/runtime/collections/Query.java]: Do we need scanAndFilter() above? Can we just have this method only?

Comment (zalokhan) [reply runtime/src/main/java/org/corfudb/runtime/collections/Query.java]: Done.

File: runtime/src/main/java/org/corfudb/runtime/collections/Table.java [hunk 3]
```diff
@@ -219,6 +213,21 @@ private void TxEnd() {
...
     }
 
+    /**
+     * Clears the table.
+     */
+    public void clear() {
+        boolean beganNewTxn = false;
+        try {
+            beganNewTxn = TxBegin();
+            corfuTable.clear();
```
Comment (pankti-m) [inline runtime/src/main/java/org/corfudb/runtime/collections/Table.java]: we will clear the table even if beganNewTxn is false..

File: runtime/src/main/java/org/corfudb/runtime/collections/Table.java [hunk 4]
```diff
@@ -236,7 +254,29 @@ int count() {
...
+     */
+    @Nonnull
+    List<CorfuStoreEntry<K, V, M>> scanAndFilterByEntry(
+            @Nonnull final Predicate<CorfuStoreEntry<K, V, M>> entryPredicate) {
+        return corfuTable.scanAndFilterByEntry(recordEntry ->
+                entryPredicate.test(new CorfuStoreEntry<>(
+                        recordEntry.getKey(),
+                        recordEntry.getValue().getPayload(),
+                        recordEntry.getValue().getMetadata())))
+                .parallelStream()
```
Comment (medhavidhawan) [inline runtime/src/main/java/org/corfudb/runtime/collections/Table.java]: Just make sure that we do not get in trouble with parallelStream as it uses common fork join pool.

Comment (zalokhan) [reply runtime/src/main/java/org/corfudb/runtime/collections/Table.java]: These are not running long-running tasks but just wrapping the entry. Also, this `parallelStream` does not run the user's predicate and hence is bounded.

**Pull request merged**